crates/aptos/src/move_tool/mod.rs

```diff
@@ -1531,8 +1531,33 @@ impl ArgWithType {
                 _ => serde_json::to_value(bcs::from_bytes::<T>(&self.arg)?)
                     .map_err(|err| CliError::UnexpectedError(err.to_string())),
             },
-            1 => serde_json::to_value(bcs::from_bytes::<Vec<T>>(&self.arg)?)
-                .map_err(|err| CliError::UnexpectedError(err.to_string())),
+            1 => match self._ty.clone() {
+                FunctionArgType::U64 => {
+                    let u64_vector: Vec<u64> = bcs::from_bytes::<Vec<u64>>(&self.arg)?;
+                    let string_vector: Vec<String> =
+                        u64_vector.iter().map(ToString::to_string).collect();
+                    serde_json::to_value(string_vector)
+                        .map_err(|err| CliError::UnexpectedError(err.to_string()))
+                },
+                FunctionArgType::U128 => {
+                    let u128_vector: Vec<u128> = bcs::from_bytes::<Vec<u128>>(&self.arg)?;
+                    let string_vector: Vec<String> =
+                        u128_vector.iter().map(ToString::to_string).collect();
+                    serde_json::to_value(string_vector)
+                        .map_err(|err| CliError::UnexpectedError(err.to_string()))
+                },
+                FunctionArgType::U256 => {
+                    let u256_vector: Vec<U256> = bcs::from_bytes::<Vec<U256>>(&self.arg)?;
+                    let string_vector: Vec<String> =
+                        u256_vector.iter().map(ToString::to_string).collect();
+                    serde_json::to_value(string_vector)
+                        .map_err(|err| CliError::UnexpectedError(err.to_string()))
+                },
+                FunctionArgType::Raw => serde_json::to_value(&self.arg)
+                    .map_err(|err| CliError::UnexpectedError(err.to_string())),
+                _ => serde_json::to_value(bcs::from_bytes::<Vec<T>>(&self.arg)?)
```

Comment:
I can't really understand why the move tests work on `u16` and `u32`, etc when there's no case that matches their type. How come `u64, u128, u256` need to be converted into string vectors first but the others don't?

Reply:
but other than that looks like a good start to me and it works :)

Reply:
@xbtmatt Yep I was actually wondering the same thing. Quoting the response I got from Greg "Numbers in JSON only are supported up to u32.  The larger ones have to be converted to a string to be supported"

You will also notice that when you run `aptos move view` and the func returns u64/u128/u256, the value returned is actually a `string` as well. e.g `"123"` instead of `123`